client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogma.java

```diff
@@ -540,10 +562,32 @@
      * watcher.watch((revision, contents) -> {
      *     ...
      * });}</pre>
-     * Note that you may get {@link RevisionNotFoundException} during the {@code getFiles()} call and
+     * {@link Function} by default is executed by a blocking task executor so that you can safely call a
+     * blocking operation.
+     *
+     * <p>Note that you may get {@link RevisionNotFoundException} during the {@code getFiles()} call and
      * may have to retry in the above example due to
      * <a href="https://github.com/line/centraldogma/issues/40">a known issue</a>.
      */
     <T> Watcher<T> repositoryWatcher(String projectName, String repositoryName, String pathPattern,
                                      Function<Revision, ? extends T> function);
+
+    /**
+     * Returns a {@link Watcher} which notifies its listeners when the specified repository has a new commit
+     * that contains the changes for the files matched by the given {@code pathPattern}. e.g:
+     * <pre>{@code
+     * Watcher<Map<String, Entry<?>> watcher = client.repositoryWatcher(
+     *         "foo", "bar", "/*.json", revision -> client.getFiles("foo", "bar", revision, "/*.json").get());
```

Comment:
nit: 
```suggestion
     *         "foo", "bar", "/*.json", revision -> client.getFiles("foo", "bar", revision, "/*.json").join(), executor);
```
We need to use `join` if we don't want to handle `InterruptedException`.
Could you also fix the other examples, please? 😄